src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -2813,6 +2813,16 @@ private static DType UnionCore(ref bool fError, DType type1, DType type2, bool u
                         isMatchingExpandType = IsMatchingExpandType(field2Type, field1Type);
                         expandType = field2Type;
                     }
+                    else if (field1Type.HasPolymorphicInfo && field2Type.IsAggregate)
```

Comment:
>HasPolymorphicInfo [](http://example.com/codeflow?start=40&length=18)

Do we need to do some additional comparison? 
Would this match Polymorphic Contacts with Aggregate Accounts?